sdcm/tester.py

```diff
@@ -662,17 +662,20 @@ def setUp(self):  # pylint: disable=too-many-branches,too-many-statements
             else:
                 self.init_nodes(db_cluster=db_cluster)
 
-            with temp_authenticator(db_cluster.nodes[0], "org.apache.cassandra.auth.PasswordAuthenticator"):
-                # running `set_system_auth_rf()` before changing authorization/authentication protocols
-                self.set_system_auth_rf(db_cluster=db_cluster)
-                if self.params.get('use_ldap'):
-                    self._setup_ldap_roles(db_cluster=db_cluster)
-                if self.params.get('ldap_server_type') == LdapServerType.MS_AD:
-                    change_default_password(node=db_cluster.nodes[0],
-                                            user=self.params.get('authenticator_user'),
-                                            password=self.params.get('authenticator_password'))
-                    self.loaders.added_password_suffix = True
-                    self.monitors.added_password_suffix = True  # FIXME: Replace with strong password generation.
+            # running `set_system_auth_rf()` before changing authorization/authentication protocols
+            self.set_system_auth_rf(db_cluster=db_cluster)
```

Comment:
taking this out of the context manager would break exactly what they are trying to fix

Reply:
Here is the description of that change:

```
This refactor fixes code duplication related to ldap setup, assures
authentication is available at all times during setup and splits
configuration into different methods, simplifying conditionals used when
configuring ldap
```

So, I don't see here the place for `break exactly what they are trying to fix`.

Reply:
This will break ms-ad tests because if we don't temporarily change authenticator to password auth, the `set_system_auth_rf` won't be able to authorize with the credentials it would use against ms-ad server

Reply:
so let's do the "context switch" only in those tests (ldap)

Reply:
another thing, why can't we `set_system_auth_rf` after all the resources are provisioned and set up? this is going to work like that any way when we complete the unified provision project (we will provision and configure resources in different step)

Reply:
If we are keeping this change, then it should be after potential context manager switch

Reply:
Done